pkg/build/controller/controller.go

```diff
@@ -288,6 +290,23 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	case kapi.PodRunning:
 		// The pod's still running
 		nextStatus = buildapi.BuildPhaseRunning
+	case kapi.PodPending:
+		if build.Status.Phase == buildapi.BuildPhasePending {
+			// When build is in pending state due to missing pull secrets for the image
+			// indicate this condition in reason.
+			if build.Status.Reason == buildapi.StatusReasonMissingPushSecret {
+				return nil
+			}
+			if secret := build.Spec.Output.PushSecret; secret != nil {
+				if bc.SecretClient.Secrets(build.Namespace).Get(secret.Name); err != nil && errors.IsNotFound(err) {
+					build.Status.Reason = buildapi.StatusReasonMissingPushSecret
+					if err := bc.BuildUpdater.Update(build.Namespace, build); err != nil {
```

Comment:
this only happens when Pod is in "pending" already (see the `case`). It will stay in pending until we acquire the secrets, then it should move to next stage... 

actually I think the @smarterclayton suggestion to `return nil` when the reason is secrets pending might cause the pod will never leave the "pending" state (because nothing will update the "reason" when the secrets become available). I think we will have to make request.